pkg/runtime/interfaces.go

```diff
@@ -30,6 +30,11 @@ const (
 	APIVersionInternal = "__internal"
 )
 
+type GroupVersioner interface {
+	VersionForGroup(group string) (unversioned.GroupVersion, bool)
```

Comment:
That comes from the preferred group.  The preferred group for the former
would be batch, the one for the later should be otherwise.  There is no
Kind in any of these code paths, there is only group preferences.  If you
register ScheduledJob first for group A in internal, it is the preferred
kind.

On Mon, May 23, 2016 at 11:22 AM, David Eads <notifications@github.com>
wrote:

> In pkg/runtime/interfaces.go
> <https://github.com/kubernetes/kubernetes/pull/26044#discussion_r64239878>
> :
>
> > @@ -30,6 +30,11 @@ const (
> >  	APIVersionInternal = "__internal"
> >  )
> >
> > +type GroupVersioner interface {
> > +	VersionForGroup(group string) (unversioned.GroupVersion, bool)
>
> I don't think this interface is rich enough for what you need it for. The
> decision about Group and GroupVersion are both tied to the originating
> GroupVersionKind. For instance:
>
> v1.batch has Job. v2alpha1.batch has Job and ScheduledJob. The decision
> of version for a given Kind depends on the kind you're talking about.
> ScheduledJob.__internal.batch should be ScheduleJob.v2alpha1.batch, but
> Job.__internal.batch should be Job.v1.batch.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26044/files/443d00fbad3785e2c68386cd4de38344f1ee21f6..9d83ae05067c2ad7bb5d3eda95d3fc3aadd77123#r64239878>
>